python_modules/libraries/dagster-powerbi/dagster_powerbi/resource.py

```diff
@@ -233,6 +234,68 @@ def build_defs(
         )
 
 
+def _workspace_data_to_cacheable_data(
+    workspace_data: PowerBIWorkspaceData,
+) -> Sequence[Mapping[str, Any]]:
+    return [
+        data.to_cached_data()
+        for data in [
+            *workspace_data.dashboards_by_id.values(),
+            *workspace_data.reports_by_id.values(),
+            *workspace_data.semantic_models_by_id.values(),
+            *workspace_data.data_sources_by_id.values(),
+        ]
+    ]
```

Comment:
Lets make this typed `record`. We should do that as a norm so we can evolve and document the serialization formats.